src/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -37,6 +37,8 @@ export const BankAccountPicker: FC<Props> = props => {
     onSetBankAccountSelection,
   } = props
 
+  const [clientSecret, setClientSecret] = useState("")
```

Comment:
so initially `clientSecret` was defined and used in `BankDebitProvider`. this component made a request to the API and fired `setClientSecret` to set the value of `clientSecret`. It did that every time it was rendered, since every time it is rendered the value of `clientSecret` was re-set to empty string (because it's defined there). 

Imagine this: a user lands on the payment page and has a previously saved account auto-selected for them. User clicks on 'Add new account', we fetch a new client secret. Then user clicks on a previously saved account and 'Add new account' again, we fetch a new client secret... this can go on. every time we fetch a secret, we are unnecessarily creating a transaction with setupIntent on the backend. 

Since `BankAccountPicker` is the parent that has all the radio buttons for selecting previously saved bank account and 'Add new account...', the value of `clientSecret` is not re-set when users switches between the radio buttons. It is now set once when user clicks on 'Add new account' only and its value is not lost as long as the component remains mounted (as long as ACH is selected as a payment method).

Reply:
for the 2nd one: I think Stripe's elements options (to which we pass the clientSecret) would complain about the potentially null property, but I'll give it a try. agreed that using null instead of empty string would be better.

Reply:
OK I could bypass it with:

```
const options = {
    clientSecret: clientSecret || "",
    ....
}
```